pyads/symbol.py

```diff
@@ -0,0 +1,265 @@
+            self.plc_type = self.get_type_from_str(self.symbol_type)
+
+    def create_symbol_from_info(self) -> None:
+        """Look up remaining info from the remote
+
+        The name must already be present.
+        Other values will already have a default value (mostly None).
+        """
+        info = adsGetSymbolInfo(self._plc._port, self._plc._adr, self.name)
+
+        self.index_group = info.iGroup
+        self.index_offset = info.iOffs
+        if info.comment:
+            self.comment = info.comment
+
+        # info.dataType is an integer mapping to a type in
+        # constants.ads_type_to_ctype.
+        # However, this type ignores whether the variable is really an array!
+        # So are not going to be using this and instead rely on the textual
+        # type
+        self.symbol_type = info.symbol_type  # Save the type as string
+
+    def read_write_check(self):
+        """Assert the current object is ready to read from/write to"""
+        if self.plc_type is None:
+            raise ValueError('Cannot read or write with invalid value for '
+                             'plc_type: `{}`'.format(self.plc_type))
+
+        if not self._plc or not self._plc.is_open:
+            raise ValueError('Cannot read or write data with missing or '
+                             'unopened Connection')
+
+        if not isinstance(self.index_group, int) or \
+                not isinstance(self.index_offset, int):
+            raise ValueError(
+                'Cannot read or write data with invalid values for group- and '
+                'offset index: ({}, {})'.format(self.index_group,
+                                                self.index_offset))
+
+    def read(self) -> Any:
+        """Read the current value of this symbol"""
+        self.read_write_check()
+        return self._plc.read(self.index_group, self.index_offset,
+                              self.plc_type)
+
+    def write(self, new_value: Any):
+        """Write a new value to the symbol"""
+        self.read_write_check()
+        return self._plc.write(self.index_group, self.index_offset,
+                               new_value, self.plc_type)
+
+    @property
+    def value(self):
+        """Equivalent to AdsSymbol.read()"""
+        return self.read()
+
+    @value.setter
+    def value(self, new_value):
+        """Equivalent to AdsSymbol.write()"""
+        self.write(new_value)
+
+    def __repr__(self):
+        """Debug string"""
+        t = type(self)
+        return '<{}.{} object at {}, name: {}, type: {}>'.format(
+            t.__module__, t.__qualname__, hex(id(self)),
+            self.name, self.symbol_type)
+
+    def __del__(self):
+        """Destructor"""
+        self.clear_device_notifications()
+
+    def add_device_notification(
+            self,
+            callback: Callable,
+            attr: Optional[NotificationAttrib] = None,
+            user_handle: Optional[int] = None
+    ) -> Optional[Tuple[int, int]]:
```

Comment:
```suggestion
    def add_device_notification(
        self,
        callback: Callable,
        attr: Optional[NotificationAttrib] = None,
        user_handle: Optional[int] = None,
    ) -> Optional[Tuple[int, int]]:
```

Reply:
Indentation should always be 4 spaces.

Reply:
You can just run black against the codebase and it should fix the indentation issues amoung others.

Reply:
Yup, I just ran black